### DISCUSSION TITLE
Support for OIDC extensions: OIDC4IA / OIDC4VP

### DISCUSSION THREAD

**Discussion by tristan-denmat-ariadnext** [keycloak/keycloak#15725] [Ideas]
Hello, 

There have been multiple initiatives recently to exend OIDC to support more KYC features.

For example _OIDC4IA_ describes how OIDC could be used to let a SP collect identity data according to specific KYC regulations: [https://openid.net/specs/openid-connect-4-identity-assurance-1_0.html]

In the same idea, _OIDC4VP_ shows how W3C Verifiable Credentials ([https://www.w3.org/TR/vc-data-model/]) could be presented by users to SP based on OIDC flows:  [https://openid.net/specs/openid-4-verifiable-presentations-1_0.html]

Have you already seen these extensions and maybe scheduled this into a roadmap ?

If not, do you see these extensions as good candidates to be included into core Keycloak or do you think it should rather be provided as plugins ?

Thanks for your feedback

**Comment by tnorimat** [comment]
@tristan-denmat-ariadnext Hello,

How about separating OIDC4IDA and OIDC4VP as each discussion?
